ethereum/core/src/main/java/tech/pegasys/pantheon/ethereum/storage/keyvalue/RocksDbStorageProvider.java

```diff
@@ -12,22 +12,83 @@
  */
 package tech.pegasys.pantheon.ethereum.storage.keyvalue;
 
+import static java.util.Arrays.asList;
+
 import tech.pegasys.pantheon.ethereum.storage.StorageProvider;
 import tech.pegasys.pantheon.metrics.MetricsSystem;
+import tech.pegasys.pantheon.services.kvstore.ColumnarRocksDbKeyValueStorage;
 import tech.pegasys.pantheon.services.kvstore.KeyValueStorage;
 import tech.pegasys.pantheon.services.kvstore.RocksDbConfiguration;
 import tech.pegasys.pantheon.services.kvstore.RocksDbKeyValueStorage;
+import tech.pegasys.pantheon.services.kvstore.SegmentedKeyValueStorage;
+import tech.pegasys.pantheon.services.kvstore.SegmentedKeyValueStorage.Segment;
+import tech.pegasys.pantheon.services.kvstore.SegmentedKeyValueStorageAdapter;
 
 import java.io.IOException;
 import java.nio.file.Files;
 
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
 public class RocksDbStorageProvider {
+  private static final Logger LOG = LogManager.getLogger();
 
   public static StorageProvider create(
       final RocksDbConfiguration rocksDbConfiguration, final MetricsSystem metricsSystem)
       throws IOException {
+    if (rocksDbConfiguration.useColumns()) {
+      return createSegmentedProvider(rocksDbConfiguration, metricsSystem);
+    } else {
+      return createUnsegmentedProvider(rocksDbConfiguration, metricsSystem);
+    }
+  }
+
+  private static StorageProvider createUnsegmentedProvider(
+      final RocksDbConfiguration rocksDbConfiguration, final MetricsSystem metricsSystem)
+      throws IOException {
     Files.createDirectories(rocksDbConfiguration.getDatabaseDir());
```

Comment:
We probably should have an info log here to indicate that we're using the original format.  Maybe even a warning that an old format is being used (not at all sure about that...)

Reply:
The warning is already an acceptance criterion of the database versioning task